core/src/main/java/org/apache/iceberg/ManifestFilterManager.java

```diff
@@ -429,13 +425,44 @@ private ManifestFile filterManifestWithDeletedFiles(
     }
   }
 
-  private Evaluator strictDeleteEvaluator(PartitionSpec spec) {
-    Expression strictExpr = Projections.strict(spec).project(deleteExpression);
-    return new Evaluator(spec.partitionType(), strictExpr);
-  }
+  // an evaluator that checks whether rows in a file may/must match a given expression
+  // this class first partially evaluates the provided expression using the partition tuple
+  // and then checks the remaining part of the expression using metrics evaluators
+  private class ExpressionEvaluator {
```

Comment:
On reservation about this: the name is really generic. What about `PartitionAndMetricsEvaluator` instead?

Reply:
Sounds good. It is an internal class so I did not worry about it. Do we want to keep it as a private inner class? The logic seems generic enough so that we can make move it into a top-level class.